src/ObjectBuilder.cs

```diff
@@ -92,12 +96,24 @@ public virtual IModelType BuildServiceType(string serviceTypeName, bool required
                             var value = valueOverride["value"];
                             var description = valueOverride["description"];
                             var name = valueOverride["name"] ?? value;
-                            enumType.Values.Add(new EnumValue
+                            
+                            var enumVal = new EnumValue
                             {
                                 Name = (string)name,
                                 SerializedName = (string)value,
                                 Description = (string)description
-                            });
+                            };
+
+                            // set the allowedValues if any
+                            if(valueOverride["allowedValues"]!=null)
+                            {
+                                foreach(var allowedValue in JArray.Parse(valueOverride["allowedValues"].ToString()))

```

Comment:
isn't `valueOverride["allowedValues"]` a `JArray` (otherwise, I wouldn't understand why `ToString` emits JSON)? I'd simply do `valueOverride["allowedValues"] is JArray allowedValues` in the `if` above, then you can iterate `allowedValues`

Reply:
neat!